flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/factories/TestValuesTableFactory.java

```diff
@@ -633,8 +648,11 @@ public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
 					sinkFunction = new KeyedUpsertingSinkFunction(
 						tableName,
 						converter,
-						keyIndices);
+						keyIndices,
+						expectedNum);
 				} else {
+					checkArgument(expectedNum == -1,
+						"Retracting Sink doesn't support '" + SINK_EXPECTED_MESSAGES_NUM.key() + "' yet.");
```

Comment:
Should we move the check to the constructor ?

Reply:
I think there is not much difference. It's more easier to check the `expectedNum` in the `getSinkRuntimeProvider`, because we have to check `runtimeSink` and `primaryKey`.